modules/lib/lib_gui.py

```diff
@@ -0,0 +1,79 @@
+import tkinter as tk
+import pyperclip
+from tkinter import messagebox
+from lib_pass_gen import PasswordGenerator
+
+# This class allows us to create entries with placeholders so it can provide more detail to use wihout additional labels.
+class EntryWithPlaceholder(tk.Entry):
+    def __init__(self, master=None, placeholder="PLACEHOLDER", textvariable=None, color="grey"):
+        super().__init__(master)
+
+        self.placeholder = placeholder
+        self.placeholder_color = color
+        self.default_fg_color = self["fg"]
+
+        if textvariable:
+            self.configure(textvariable=textvariable)
+
+        self.bind("<FocusIn>", self.foc_in)
+        self.bind("<FocusOut>", self.foc_out)
+
+        self.put_placeholder()
+
+    def put_placeholder(self):
+        self.insert(0, self.placeholder)
+        self["fg"] = self.placeholder_color
+
+
+
+    def foc_in(self, *args):
+        if self["fg"] == self.placeholder_color:
+            self.delete("0", "end")
+            self["fg"] = self.default_fg_color
+
+    def foc_out(self, *args):
+        if not self.get():
+            self.put_placeholder()
+
+
+# I assume user will need to type in a number in few modules so this creating entry with validation if input is stricly number may be handy
+def is_number(user_input):
+    if user_input.get().isdigit():
```

Comment:
Isn't this function's body the same as just:
```python
def is_number(user_input):
    return user_input.get().isdigit()
```

Reply:
it's the same, and I moved it to utils.py with copy_to_clipboard() function

Reply:
See here types in python would help us. the user_input is not a generic string but a tkinter string. This is not a generic is_number function then and having it in the utils makes no sense. Leave it here